mailbox/store/src/main/java/org/apache/james/mailbox/store/search/ListeningMessageSearchIndex.java

```diff
@@ -60,17 +63,24 @@ public ListeningMessageSearchIndex(MessageMapperFactory factory) {
     protected MessageMapperFactory getFactory() {
         return factory;
     }
-    
-    
+
+    /**
+     * Return the {@link MailboxManager}
+     *
+     * @return mailboxManager
+     */
+    public MailboxManager getMailboxManager() {

```

Comment:
it looks like a bad idea

Reply:
this method is only used inside `LazyMessageSearchIndex`, that's all
```java
    private final ListeningMessageSearchIndex index;

    public LazyMessageSearchIndex(ListeningMessageSearchIndex index) {
        super(index.getFactory(), index.getMailboxManager());
        this.index = index;
    }
```

Reply:
No. It's a public method returning a internal object information for working around a badly designed class. And now that's public, you don't manage what part of the code can use it.

What about trying to fix the root cause instead ?

Reply:
I rework a bit of `MessageSearchIndex`, please checkout https://github.com/linagora/james-project/pull/1980/commits/5c7d51653210dedf533689847ba471fdda1c4395